File: src/main/java/ru/ssau/tourism/entities/Season.java
package ru.ssau.tourism.entities;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Setter
@NoArgsConstructor
@EqualsAndHashCode(of = {"id"})
public class Season {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "tour_id")
    private Tour tour;
    @Column
    private LocalDate startDate;
    @Column
    private LocalDate finishDate;
    @Column
    private Boolean isFinished;
    @Column
    private Integer countOfPlaces;

    public Long getId() {
        return id;
    }

    public Tour getTour() {
        return tour;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public LocalDate getFinishDate() {
        return finishDate;
    }

    public Boolean getFinished() {
        return isFinished;
    }

    public Integer getCountOfPlaces() {
        return countOfPlaces;
    }
}

File: src/main/kotlin/ru/ssau/tourism/entities/Season.kt
package ru.ssau.tourism.entities

import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDate
import javax.persistence.*

@Entity
class Season {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	var id: Long? = null
	@ManyToOne
	@JoinColumn(name = "tour_id")
	var tour: Tour? = null
	@Column
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	var startDate: LocalDate? = null
	@Column
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	var finishDate: LocalDate? = null
	@Column
	var finished: Boolean? = false
	@Column
	var countOfPlaces: Int? = null
}